src/tools/spin/spinquic.cpp

```diff
@@ -200,6 +201,11 @@ QUIC_STATUS QUIC_API SpinQuicHandleStreamEvent(HQUIC Stream, void * /* Context *
     case QUIC_STREAM_EVENT_PEER_SEND_SHUTDOWN:
         MsQuic->StreamShutdown(Stream, (QUIC_STREAM_SHUTDOWN_FLAGS)GetRandom(16), 0);
         break;
+    case QUIC_STREAM_EVENT_RECEIVE:
+        if (GetRandom(5) == 0) {
```

Comment:
Let's add another `if (GetRandom(5) == 0)` condition after this to consume only part of the data (i.e. reduce `TotalBufferLength` by some value).

Reply:
Never mind. I was confused by what you were doing below. This can wait until the next PR.